### PR TITLE
OpenBSD build fix proposal

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: zutil.h [hunk 1]
```diff
@@ -185,9 +185,15 @@ void Z_INTERNAL   zng_cfree(void *opaque, void *ptr);
 
 #elif defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
 #  include <sys/endian.h>
+#  if defined(__OpenBSD__)
+#  define ZSWAP16(q) __builtin_bswap16(q)
+#  define ZSWAP32(q) __builtin_bswap32(q)
+#  define ZSWAP64(q) __builtin_bswap64(q)
```
Comment (mtl1979) [inline zutil.h]: These should be `swap16`, `swap32` and `swap64` according to OpenBSD headers... We should not use compiler builtins if functions exist in standard headers. Might warrant separating OpenBSD as own `#elif` block.

Comment (devnexen) [reply zutil.h]: Yes there are but tried to use intrinsics rather than the macros (might be slower) but nevermind I ll change.

Comment (mtl1979) [reply zutil.h]: At least gcc should map the functions in the standard headers automatically to the intrinsics when optimizations are enabled.

Comment (devnexen) [reply zutil.h]: true

**Review by mtl1979 — APPROVED** [review]

**Pull request merged**